src/main/java/core/basesyntax/dao/ma/CoachDaoImpl.java

```diff
@@ -11,6 +13,11 @@ public CoachDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public List<Coach> findByExperienceGreaterThan(int years) {
-        return null;
+        try (Session session = sessionFactory.openSession()) {
```

Comment:
I would suggest to use CriteriaQuery as its created for this purpose (for select queries)

Reply:
@jah701 you can equally use HQL for select queries